solr/contrib/blob-directory/src/java/org/apache/solr/blob/BlobDirectory.java

```diff
@@ -56,10 +61,22 @@
   private final Object lock = new Object();
   private volatile boolean isOpen;
 
-  public BlobDirectory(Directory delegate, String blobDirPath, BlobPusher blobPusher) {
+  public BlobDirectory(Directory delegate, String blobDirPath, BlobStoreConnection blobStoreConnection) throws IOException {
     super(delegate);
     this.blobDirPath = blobDirPath;
-    this.blobPusher = blobPusher;
+    this.blobStoreConnection = blobStoreConnection;
+    pullMissingFilesFromRepo();
+  }
+
+  private void pullMissingFilesFromRepo() throws IOException {
+    Set<String> localFileNames = new HashSet<>(Arrays.asList(in.listAll()));
+    blobStoreConnection.pull(blobDirPath, this::openOutput, blobFile -> {
```

Comment:
If locally we're empty and we pull down files, then we should have a INFO log message saying that.
On the other hand if there's no transfer or deletions to be done, we should not log anything.